core/server/common/src/main/java/alluxio/master/journal/checkpoint/OptimizedCheckpointOutputStream.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal.checkpoint;
+
+import alluxio.util.FormatUtils;
+
+import net.jpountz.lz4.LZ4FrameOutputStream;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.security.DigestOutputStream;
+import java.security.MessageDigest;
+
+/**
+ * OutputStream to write checkpoint files efficiently.
+ */
+public class OptimizedCheckpointOutputStream extends OutputStream {
+  public static final int BUFFER_SIZE = (int) FormatUtils.parseSpaceSize("4MB");
```

Comment:
Actually maybe a smaller buffer size is sufficient, we can do some performance testing

Reply:
Did some testing and buffer size does not seem to affect performance in our case.